drivers/wifi/nxp/Kconfig.nxp

```diff
@@ -0,0 +1,1184 @@
+	int "Wlcmgr thread stack size"
+	default 5120
+	help
+	  This option specifies the size of the stack used by the wlcmgr task.
+
+config NXP_WIFI_POWERSAVE_TASK_STACK_SIZE
+	int "Wifi powersave task stack size"
+	default 512
+	help
+	  This option specifies the size of the stack used by the wifi powersave task.
+
+config NXP_WIFI_TX_TASK_STACK_SIZE
+	int "Wifi driver TX task stack size"
+	default 2048
+	depends on NXP_WIFI_WMM
+	help
+	  This option specifies the size of the stack used by the wifi driver TX task.
+
+config NXP_WIFI_DRIVER_TASK_STACK_SIZE
+	int "Wifi driver task stack size"
+	default 2048
+	help
+	  This option specifies the size of the stack used by the wifi driver task.
+
+config NXP_WIFI_SCAN_TASK_STACK_SIZE
+	int "Wifi scan task stack size"
+	default 2048
+	help
+	  This option specifies the size of the stack used by the wifi scan task.
+
+endmenu
+
+menu "Wi-Fi Station Support"
+
+config NXP_WIFI_STA_AUTO_CONN
+	bool "Automatically starts STA connection"
+	default n
+	help
+	  This option enables Station auto connection support.
+
+if NXP_WIFI_STA_AUTO_CONN
+
+config NXP_WIFI_STA_AUTO_SSID
+	string "NXP Wi-Fi STA auto connect SSID"
+	depends on NXP_WIFI_STA_AUTO_CONN
+	default "myssid"
+	help
+	  This option specifies the SSID of the external Access Point
+	  for auto connection.
+	  Maximum length is 32 ascii characters.
+
+config NXP_WIFI_STA_AUTO_PASSWORD
+	string "NXP Wi-Fi STA auto connect password"
+	depends on NXP_WIFI_STA_AUTO_CONN
+	default ""
+	help
+	  This option specifies the Passphrase of the external
+	  Access Point for auto connection.
+	  Only support PSK security or open security by default.
+	  Length range 8 - 64, or 0 for open security.
+
+endif
+
+config NXP_WIFI_STA_RECONNECT
+	bool "Automatically starts STA Reconnection"
+	default y
+	help
+	  This option enables Station auto reconnection support,
+	  when disconnected from current associated Access Point.
+
+config NXP_WIFI_AUTO_POWER_SAVE
+	bool "Automatically starts Power Save support"
+	default y
+	help
+	  WiFi driver will automatically initiate power save when initialized.
+
+config NXP_WIFI_STA_AUTO_DHCPV4
+	bool "Automatically starts DHCPv4 negotiation"
+	default y
+	depends on NET_DHCPV4
+	depends on NET_IPV4
+	depends on NXP_WIFI_IP_DHCP
+	help
+	  WiFi driver will automatically initiate DHCPV4 negotiation when connected.
+
+choice
+	prompt "NXP IP Address configuration"
+	default NXP_WIFI_IP_DHCP
+	help
+	  Choose whether to use an IP assigned by DHCP Server or
+	  configure a static IP Address.
+
+config NXP_WIFI_IP_DHCP
+	bool "DHCP"
+	help
+	  Use DHCP to get an IP Address.
+
+config NXP_WIFI_IP_STATIC
+	bool "Static"
+	help
+	  Setup Static IP Address.
+
+endchoice
+
+if NXP_WIFI_IP_STATIC
+
+config NXP_WIFI_IP_ADDRESS
+	string "NXP Station mode IP Address"
+
+config NXP_WIFI_IP_GATEWAY
+	string "Gateway Address"
+
+config NXP_WIFI_IP_MASK
+	string "Network Mask"
+	help
+	  This option set net mask, such as 255.255.255.0.
+endif
```

Comment:
Same for these, the IP address configuration is related to global network configuration and not related to wifi driver.